src/Host.cpp

```diff
@@ -456,6 +464,22 @@ void Host::reenableAllTriggers()
     mTimerUnit.reenableAllTriggers();
 }
 
+QPair<QString, QString> Host::getSearchEngine()
+{
+    auto ret = QPair<QString, QString>();
+    ret.first = QString("Google");
+    ret.second = mSearchEngineData[ret.first];
+
+    auto iterator = mSearchEngineData.constFind(mSearchEngineName);
+    if(iterator != mSearchEngineData.end())
+    {
+        ret.first = iterator.key();
+        ret.second = iterator.value();
+    }
+
+    return ret;
```

Comment:
I think it is silly to iterate ourself through the `QMap`, Qt is perfectly capable of doing it for us with the `const T QMap::value(const Key &key, const T &defaultValue = T()) const` method, so this all can be simplified to:

    if (mSearchEngineData.contains(mSearchEngineName)) {
        return qMakePair(mSearchEngineName, mSearchEngineData.value(mSearchEngineName));
    } else {
        return qMakePair(QStringLiteral("Google"), mSearchEngineData.value(QStringLiteral("Google")));
    }

Of course if it is only required to have the relevant value from the `(QMap<QString, QString>) mSearchEngineData` then it could be:

    return mSearchEngineData.value(mSearchEngineName, mSearchEngineData.value(QStringLiteral("Google")));